### DISCUSSION TITLE
Help with rule to turn Sonoff off nd then on after a delay

### DISCUSSION THREAD

**Discussion by kds54** [arendst/Tasmota#21307] [Q&A] [answered]
A remote location with a Raspberry Pi and an internet connection that can become flaky.  I'm trying to set a rule in a Sonoff S31 running Tasmota 13.4.0 so that if the Pi does not have an internet connection it sends a command such as <<http://192.168.1.35/cm?cmnd=PowerCycle>> to turn off the Sonoff at the router and then turn it back on after a 10 second delay.  I've read through the rules information but they seem to be way more complicated than what I need and haven't been successful in finding a discussion leading to what i want.  I know this has to be simple but I'm stumped.

I've tried several different rules and here is my latest:
Rule1 ON "PowerCycle" DO Backlog POWER TOGGLE; Delay 600; POWER TOGGLE ENDON | Break
With everything I've tried, I get <<Command:  "Unknown">>

Can someone please help me to get the syntax correct?  TIA

**Comment by sfromis** [comment, marked as answer]
First, you need no rules alt all for this. With current Tasmota, this new command does what you suggest, turning off and then on again after 10 seconds (10000 milliseconds):
`TimedPower 10000,off`

When it comes to rules, you cannot create new commands, trigger expressions (between `on` and `do`) are only for very specific constructs. You could have the command `Event PowerCycle` matching a rule trigger like `on Event#PowerCycle do ... endon`. This could be used to invoke the `TimedPower` command, but you could also just issue this command directly without going via a rule.

Also, that `|` is not valid syntax, and you should normally end each rule with `endon` instead of `break`, the latter not being useful in your case.

**Reply by kds54** [reply]
Thanks so much for your help!  I knew that it had to be something simple but I couldn't figure it out.  Your suggestion works like a charm.
